homework/list.js

```diff
@@ -52,28 +44,43 @@
       buff.appendChild(li);
     });
     ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
+    this._wrapper.appendChild(ul);
+  },
 
-  function fetchList(afterFetch) {
+  fetchList(afterFetch) {
+	return new Promise((function(resolve,reject){
     var xhr = new XMLHttpRequest();
-    xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
+    //xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
+    xhr.open('GET', 'demo-list-notes.json', true);
     xhr.responseType = 'json';
     xhr.onreadystatechange = function(e) {
       // Watch out: we have a mysterious unknown 'this'.
       if (this.readyState === 4 && this.status === 200) {
         var listData = this.response;
         // The flow ends here.
-        afterFetch(listData);
+		resolve(listData);
       } else if (this.status !== 200 ){
         // Ignore error in this case.
+		reject('FETCHING FAILED: ' + this.status + ' ' + this.readyState);
       }
     };
     xhr.send();
-  }
-
-  document.addEventListener('DOMContentLoaded', function(event) {
-    start();
-  });
+	}).bind(this));
+  },
+  
+    start() {
+	this._wrapper = document.querySelector('#note-list-wrapper');
+    this.fetchList()
```

Comment:
You should practice to compose a promise chain with multiple `then` and `catch`. And you should **log AND re-throw** the error in the `catch`, to prevent the following chains make the error silent and not being interrupted.